stdlib/http/src/main/ballerina/http/http_secure_client.bal

```diff
@@ -294,178 +345,603 @@ public function createHttpSecureClient(string url, ClientEndpointConfig config)
-                var jwtIssuerConfig = authConfig["inferredJwtIssuerConfig"];
-                if (jwtIssuerConfig is ()) {
-                    authToken = runtime:getInvocationContext().authenticationContext.authToken;
-                } else {
-                    auth:JwtHeader header = { alg: jwtIssuerConfig.signingAlg, typ: "JWT" };
-                    auth:JwtPayload payload = {
-                        sub: runtime:getInvocationContext().principal.username,
-                        iss: jwtIssuerConfig.issuer,
-                        exp: time:currentTime().time / 1000 + jwtIssuerConfig.expTime,
-                        iat: time:currentTime().time / 1000,
-                        nbf: time:currentTime().time / 1000,
-                        jti: system:uuid(),
-                        aud: jwtIssuerConfig.audience
-                    };
-                    auth:JWTIssuerConfig issuerConfig = {
-                        keyStore: jwtIssuerConfig.keyStore,
-                        keyAlias: jwtIssuerConfig.keyAlias,
-                        keyPassword: jwtIssuerConfig.keyPassword
-                    };
-                    var token = auth:issueJwt(header, payload, issuerConfig);
-                    // TODO: cache the token per-user per-client and reuse it
-                    if (token is string) {
-                        authToken = token;
-                    } else {
-                        return token;
-                    }
-                }
+            if (authConfig is JwtAuthConfig) {
+                authToken = check getAuthTokenForJWTAuth(authConfig);
+            } else if (authConfig is OAuth2AuthConfig || authConfig is BasicAuthConfig) {
+                string errMsg = "JWT auth config not provided.";
+                log:printError(errMsg);
+                error e = error(HTTP_ERROR_CODE, { message: errMsg });
+                return e;
             } else {
                 authToken = runtime:getInvocationContext().authenticationContext.authToken;
             }
             if (authToken == EMPTY_STRING) {
-                error err = error(HTTP_ERROR_CODE, { message: "JWT was not used during inbound authentication.
-                                                     Provide InferredJwtIssuerConfig to issue new token." });
-                return err;
+                string errMsg = "JWT was not used during inbound authentication.
+                    Provide InferredJwtIssuerConfig to issue new token.";
+                log:printError(errMsg);
+                error e = error(HTTP_ERROR_CODE, { message: errMsg });
+                return e;
             }
             req.setHeader(AUTH_HEADER, AUTH_SCHEME_BEARER + WHITE_SPACE + authToken);
+            return false;
         } else {
-            error err = error(HTTP_ERROR_CODE, { message: "Unsupported auth scheme" });
-            return err;
+            string errMsg = "Unsupported auth scheme.";
```

Comment:
How can we get into this else clause?  Can auth.scheme can have anything other than BASIC_AUTH,OAUTH2 or JWT_AUTH

Reply:
Yes, we can't. Earlier the `scheme` was a `string` type variable. The else block is propagated from those days. Now we do not need it since `scheme` should be matched to one of `http:BASIC_AUTH`, `http:OAUTH2` or `http:JWT_AUTH` from compile time.